src/staticarrays_support.jl

```diff
@@ -8,12 +8,19 @@ The `staticschema` of a `StaticArray` element type is the `staticschema` of the
 julia> StructArrays.staticschema(SVector{2, Float64})
 Tuple{Float64, Float64}
 ```
+The one exception to this rule is `<:StaticArrays.FieldArray`, since `FieldArray` is based on a 
+struct. In this case, `staticschema(<:FieldArray)` returns the `staticschema` for the struct 
+which subtypes `FieldArray`. 
 """
 @generated function StructArrays.staticschema(::Type{<:StaticArray{S, T}}) where {S, T}
     return quote
         Base.@_inline_meta
         return NTuple{$(tuple_prod(S)), T}
     end
 end
+# invoke the general fallback for a `FieldArray` type.
+@inline function StructArrays.staticschema(T::Type{<:FieldArray})
+    invoke(StructArrays.staticschema, Tuple{Type{<:Any}}, T)
+end
 StructArrays.createinstance(::Type{T}, args...) where {T<:StaticArray} = T(args)
 StructArrays.component(s::StaticArray, i) = getindex(s, i)
```

Comment:
Actually, we also need to change this, in case users were using a custom `getindex` (this is the "mismatching for non column major order" issue I was mentioning before). So I think `component(s::FieldArray)` should also be special-cased to call `getfield`.

Reply:
Thanks for the clarification. 

Just to be clear, is this meant to enforce that `StructArrays.component(s::FieldArray, i)` returns the first field of the `FieldVector` struct rather than first index of the `FieldVector`? It seems to do this even without specializing `component` for `FieldVectors`.

Reply:
For context, I'm trying to write a test for the custom index case...

Reply:
So, I think the best check is to write something like

```julia
julia> using StaticArrays

julia> struct FlippedVec2D <: FieldVector{2,Float64}
           x::Float64
           y::Float64
       end

julia> Base.getindex(f::FlippedVec2D, i::Int) = ifelse(i == 1, f.y, f.x)

julia> Base.Tuple(f::FlippedVec2D) = (f.y, f.x)
```

And make sure that when you do eg `StructArray([FlippedVec2d(0.0, 1.0)])` you get the correct thing (`s.x[1] == s[1].x`). With the PR as is, I think it would get it wrong.

Reply:
It actually gives the right answer even without specializing `component`, though the constructor `StructArray([FlippedVec2d(0.0, 1.0)])` fails if `component` is not specialized to `FieldArray`.